python/pyarrow/array.pxi

```diff
@@ -285,7 +285,12 @@ cdef class Array:
 
         return pyarrow_wrap_array(result)
 
-    def to_pandas(self):
+    def cast(self, DataType dtype):
+        cdef shared_ptr[CArray] c_array
+        check_status(self.ap.cast(dtype.sp_type, &c_array))
+        return pyarrow_wrap_array(c_array)
```

Comment:
How about `to_dictionary`, incorporating the discussion above (this will be a no-op if it is already DictionaryArray)